cranelift/codegen/src/opts.rs

```diff
@@ -78,33 +87,29 @@ impl<'a, 'b> generated_code::Context for IsleContext<'a, 'b> {
         None
     }
 
-    fn pure_enodes_etor(
-        &mut self,
-        eclass: Id,
-        multi_index: &mut usize,
-    ) -> Option<(Type, InstructionImms, IdArray)> {
-        let nodes = self.egraph.egraph.enodes(eclass);
-        while *multi_index < nodes.len() {
-            let i = *multi_index;
-            *multi_index += 1;
-            match nodes[i] {
-                Node::Pure { op, args, types } if types.len() == 1 => {
-                    let ty = types.as_slice(&self.egraph.node_ctx.types)[0];
-                    return Some((ty, op.clone(), args.clone()));
-                }
-                _ => {}
-            }
-        }
-        None
+    type pure_enodes_etor_iter = PureNodesEtorIter<'a, 'b>;
```

Comment:
This looks unused ? (was it the reason for also allowing non_camel_case_types ?)